calico.go

```diff
@@ -156,10 +156,12 @@ func cmdAdd(args *skel.CmdArgs) error {
 			// See CNI Spec doc for more details.
 			result, err = current.NewResultFromResult(ipamResult)
 			if err != nil {
+				ReleaseIPAllocation(logger, conf.IPAM.Type, args.StdinData)
```

Comment:
How much thought have you put into checking all the places where this should be called?